mesonbuild/compilers.py

```diff
@@ -2729,3 +2733,135 @@ def unix_compile_flags_to_native(self, args):
 
     def get_link_debugfile_args(self, targetfile):
         return []
+
+
+# Only tested on linux for ICC 17.0.0
+#    * object extraction does not compile (don't know) (25 object extraction fails)
+#    * ICC returns no error if not existing flags are used (i.e. -fsomeflag),
+#      therefore some tests fail  (112 has arg and test_vala_c_warning fails, and 1 vala cwerror successes)
+#    * ICC 17.0.0 does not support c++03 and g++03 (94 default options fails)
+class IntelCompiler:
+    def __init__(self, icc_type):
+        self.id = 'intel'
+        self.icc_type = icc_type
+        self.base_options = ['b_pch', 'b_lto', 'b_pgo', 'b_sanitize', 'b_coverage',
+                             'b_colorout', 'b_ndebug', 'b_staticpic', 'b_lundef', 'b_asneeded']
+        # Assembly 
+        self.can_compile_suffixes.add('s')
+
+    def get_pic_args(self):
+        return ['-fPIC']
+
+    def get_buildtype_args(self, buildtype):
+        return gnulike_buildtype_args[buildtype]
+
+    def get_buildtype_linker_args(self, buildtype):
+        return gnulike_buildtype_linker_args[buildtype]
+
+    def get_pch_suffix(self):
+        return 'pchi'
+
+    def get_pch_use_args(self, pch_dir, header):
+        return ['-pch', '-pch_dir', os.path.join(pch_dir), '-include', header]
+
+    def get_pch_name(self, header_name):
+        return os.path.split(header_name)[-1] + '.' + self.get_pch_suffix()
+
+    # Not required, but possible if backend should create pch files. However Intel does it automatically
+    #def gen_pch_args(self, header, source, pchname):
+    #    return ['-pch_create', os.path.join(pchname, self.get_pch_name(header)), '-include', header, source])
+
+    def split_shlib_to_parts(self, fname):
+        return (os.path.split(fname)[0], fname)
+
+    def get_soname_args(self, prefix, shlib_name, suffix, path, soversion, is_shared_module):
+        if self.icc_type == CLANG_STANDARD:
+            gcc_type = GCC_STANDARD
+        elif self.icc_type == CLANG_OSX:
+            gcc_type = GCC_OSX
+        elif self.icc_type == CLANG_WIN:
+            gcc_type = GCC_MINGW
+        else:
+            raise MesonException('Unreachable code when converting clang type to gcc type.')
+        return get_gcc_soname_args(gcc_type, prefix, shlib_name, suffix, path, soversion, is_shared_module)
+
+    def get_std_shared_lib_link_args(self):
+        return ['-shared']
+
+    def get_std_shared_lib_link_args(self):
```

Comment:
__[flake8]__
* [F811] redefinition of unused 'get_std_shared_lib_link_args' from line 2788

<!-- Comment by [SideCI](https://sideci.com) -->